Implement storage service monitor endpoints

jraddaoui opened the pull request:
- Update storage service events:

  - Remove location and AIP updated events.
  - Add AIP status and AIP location updated events.
  
- Publish storage service events:

  - Set up the storage event service using Redis.
  - Handle related AIP/Workflow UUIDs.
  - Add internal type to Goa type transformations.
  - Publish events from related methods.
  
- Implement storage service monitor endpoints:

  - Monitor request:
    - Stores a ticket and the user claims and returns a cookie.
  - Monitor:
    - Validates the ticket cookie and populates the user claims.
    - Subscribes to the storage event service.
    - Sends a stream of events based on user attributes.
  - Also, update ingest service monitor endpoints to:
    - Return internal errors.
    - Consider send errors as client disconnections.
    
- Handle storage service events in dashboard:

  - Connect to the storage service API WebSocket monitor endpoint.
  - Handle storage service events based on type.
  - Add fetch AIPs and fetch locations debounced methods to stores.
  - Remove polling process to update current AIP based on the status.
  - Update Dockerfile NGINX configuration and Vite server.

- Make Task component reactive to note updates:

  Replace Card class with computed properties to ensure the component
  properly updates when task.note changes. The previous implementation
  used a non-reactive class instance that wouldn't respond to prop
  changes from WebSocket events or API updates.

  - Remove Card class in favor of reactive computed properties.
  - Use computed() for note splitting logic to track prop changes.
  - Convert isOpen state to simple ref() for better reactivity.
  - Preserve expand/collapse state independently of note updates.


Refs #1222, depends on #1302.

jraddaoui committed:
Update storage service events

- Remove location and AIP updated events.
- Add AIP status and AIP location updated events.

jraddaoui committed:
Publish storage service events

- Set up the storage event service using Redis.
- Handle related AIP/Workflow UUIDs.
- Add internal type to Goa type transformations.
- Publish events from related methods.

jraddaoui committed:
Handle storage service events in dashboard

- Connect to the storage service API WebSocket monitor endpoint.
- Handle storage service events based on type.
- Add fetch AIPs and fetch locations debounced methods to stores.
- Remove polling process to update current AIP based on the status.
- Update Dockerfile NGINX configuration and Vite server.

jraddaoui committed:
Make Task component reactive to note updates

Replace Card class with computed properties to ensure the component
properly updates when task.note changes. The previous implementation
used a non-reactive class instance that wouldn't respond to prop
changes from WebSocket events or API updates.

- Remove Card class in favor of reactive computed properties.
- Use computed() for note splitting logic to track prop changes.
- Convert isOpen state to simple ref() for better reactivity.
- Preserve expand/collapse state independently of note updates.

sevein commented on internal/ingest/monitor.go:
```diff
@@ -43,21 +41,24 @@ func (w *goaWrapper) Monitor(
 	if err := stream.Send(&goaingest.IngestEvent{IngestValue: event}); err != nil {
-		return err
+		// Consider send errors as client disconnections.
+		w.logger.V(1).Info("failed to send hello event", "err", err)
```
Remember that `Info` takes a proper English sentence, i.e. it should read `"It failed to send hello event."`.

sevein reviewed (approved):
LGTM

jraddaoui committed:
Implement storage service monitor endpoints

Monitor request:
- Stores a ticket and the user claims and returns a cookie.

Monitor:
- Validates the ticket cookie and populates the user claims.
- Subscribes to the storage event service.
- Sends a stream of events based on user attributes.

Also, update ingest service monitor endpoints to:
- Return internal errors.
- Consider send errors as client disconnections.
